hwilib/commands.py

```diff
@@ -151,25 +151,24 @@ def getkeypool(args, client):
 
     # Get the key at the base
     base_key = client.get_pubkey_at_path(path_base)['xpub']
-    parent_pk, parent_cc = decompose_xpub(base_key)
 
     import_data = []
-    for i in range(start, end + 1):
-        this_import = {}
-        if (path_base[-1] == '/'):
-            path = path_base + str(i)
-        else:
-            path = path_base + '/' + str(i)
-
-        child_pk, child_cc = CKDpub(parent_pk, parent_cc, i)
-        address = pubkey_to_address(child_pk, args.testnet)
-
-        this_import['pubkeys'] = [{binascii.hexlify(child_pk).decode() : {master_fpr : path.replace('\'', 'h')}}]
-        this_import['scriptPubKey'] = {'address' : address}
-        this_import['timestamp'] = 'now'
-        this_import['internal'] = internal
-        this_import['keypool'] = keypool
-        import_data.append(this_import)
+    this_import = {}
+
+    descriptor_open = 'pkh('
+    descriptor_close = ')'
+    if args.bech32 == True:
+          descriptor_open = 'wpkh('
+    elif args.p2sh_p2wpkh == True:
+          descriptor_open = 'sh(pkh('
+          descriptor_close = '))'
+
+    this_import['desc'] = descriptor_open + '[' + master_fpr[-8:] + '/' + path_base.replace('\'','h').replace('m/', '') + ']' + base_key.replace('\'','h') + '/' + path_suffix + descriptor_close
```

Comment:
This is wrong. `master_fpr` has the fingerprint in reversed byte order due to the weirdness with uint160's. With descriptors, they should be put in the correct order as descriptors does not have that issue. You should also change the helper function to just give the fingerprint and not a uint160 so we don't need the `[-8:]` here.

Reply:
Fixed